src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -856,14 +856,17 @@ public void recalculateIsland(@NotNull Island island) {
     private void recalculateIsland(@NotNull Island island, @NotNull List<Chunk> chunks) {
         chunks.stream().map(chunk -> chunk.getChunkSnapshot(true, false, false)).forEach(chunk -> {
             World world = Bukkit.getWorld(chunk.getWorldName());
+            boolean ignoreMainMaterial = IridiumSkyblock.getInstance().getChunkGenerator().ignoreMainMaterial();
             int maxHeight = world == null ? 255 : world.getMaxHeight() - 1;
+
             for (int x = 0; x < 16; x++) {
                 for (int z = 0; z < 16; z++) {
                     if (island.isInIsland(x + (chunk.getX() * 16), z + (chunk.getZ() * 16))) {
                         final int maxy = Math.min(maxHeight, chunk.getHighestBlockYAt(x, z));
                         for (int y = LocationUtils.getMinHeight(world); y <= maxy; y++) {
                             XMaterial material = XMaterial.matchXMaterial(chunk.getBlockType(x, y, z));
-                            if (material.equals(XMaterial.AIR)) continue;
+                            if (material == XMaterial.AIR) continue;
```

Comment:
Is this a bug? think this line should be removed no?

What if its an ocean world and somone decided to drain some of the ocean, or build a base underwater or whatever. we want to regenerate that part back to water/lava no?

Reply:
This is the island recalculation method and has nothing to do with the world generator. 

Took me a few minutes to figure it out, tho xd

Reply:
oh thought it was regenerate method mb